src/HappyConnOpener.cc

```diff
@@ -325,6 +327,7 @@ HappyConnOpenerAnswer::~HappyConnOpenerAnswer()
 
 HappyConnOpener::HappyConnOpener(const ResolvedPeers::Pointer &dests, const AsyncCall::Pointer &aCall, HttpRequest::Pointer &request, const time_t aFwdStart, int tries, const AccessLogEntry::Pointer &anAle):
     AsyncJob("HappyConnOpener"),
+    codeContext(CodeContext::Current()),
```

Comment:
This code is not wrong and is even a bit faster, but please move the new codeContext field into HappyConnOpener::spareWaiting and reset it inside HappyOrderEnforcer::enqueue(). I want to emphasize that HappyOrderEnforcer is a job-independent service, and services should manage context using service-specific storage. HappyConnOpener::spareWaiting is an (invasive) HappyOrderEnforcer-dedicated storage.

Reply:
Done.